backend/core/serializers.py

```diff
@@ -499,6 +498,30 @@ class Meta:
 
 
 class RequirementAssessmentWriteSerializer(BaseModelSerializer):
+    def validate_score(self, value):
+        framework = self.get_framework()
+
+        if value is not None:
+            if value < framework.min_score or value > framework.max_score:
+                raise serializers.ValidationError(
+                    f"Score must be between {framework.min_score} and {framework.max_score}"
```

Comment:
We need to let `{score: f"...}` at the beginning to indicate field error on frontend